src/Statistics.jl

```diff
@@ -75,15 +75,15 @@ between two iterates is small. For more stopping criteria check the
 
 The algorithm is further described in [[Afsari et al, 2013]](#Afsari2013).
 """
-mean
+mean(::Manifold, args...)
 
 @doc doc"""
     mean!(M::Manifold, y, x::AbstractVector, [w::AbstractWeights]; kwargs...)
     mean!(M::Manifold, y, x::AbstractVector, [w::AbstractWeights], method::AbstractMethod; kwargs...)
 
 Compute the [`mean`](@ref) in-place in `y`.
 """
-mean!
+mean!(::Manifold, args...)
```

Comment:
Is this a `nothing`-definition and why? Can the documentation also be attached to line 88f?

Reply:
We use a definition with args as opposed to what it was before so that

```julia
julia> M = MyManifold()
? mean!(M, y, x, w)
```
gives this specific help menu, not the generic one for `Statistics.mean!`. This is even more useful for specific manifolds (e.g. see the `mean(::Sphere, args...)` case).

I don't attach it to one of the below functions for cleanliness. It defines the general interface for the user, but placing it below made it harder for me the developer to follow the logic of the different `mean` and `mean!` calls because it interrupted their flow. I don't know of any problems that this would cause, but if there are any, I can change it.

Reply:
hm, interesting. Until now that is not consistently implemented that way, but it seems reasonable to do so (maybe better in `ManifoldsBase` for most functions). Thanks for the explanation. It can of course stay that way then.

Reply:
Yes I have been learning recently about tricks like this that are very useful.

I also am adopting a convention for `mean` that maybe makes sense to adopt everywhere. Namely, our pattern is to overload the mutating functions, which changes the defaults for the non-mutating functions that call them. However, most users will probably initially implement a function with mutating and then switch to non-mutating to make more efficient. If we only document the non-mutating overload, then help messages aren't useful. But if we document both overloads, we have a lot more work to make docstrings consistent. Instead, we can adopt the convention that when we overload a mutating default, we always only document the corresponding non-mutating function, unless the non-mutating is also overloaded for the same types.

See e.g. the `mean(::Sphere, args...)` case. Here I overloaded `mean!(::Sphere, args...; kwargs...)`, but I only overloaded the docstring for `mean(::Sphere, args...)`.

Reply:
This sounds neat, although a single docstring for both mutating and non-mutating version would be a bit more complicated (it has to cover both possibilities).

Reply:
I think it's not meant to represent both; as far as I understand doing a doc-string for `mean(::Sphere, args...)` repeating the formula (from the `mean!`) provides a doctoring for the mean (non mutating) on the Sphere (which you have anyways through our `ManifoldsBase`-fallback from `mean` to `mean!`). I think the idea for user experience is nice (to type `? mean(::Sphere)` to see the doc), but in documentation it a little redundant.